src/NServiceBus.Core/Recoverability/SecondLevelRetries/SecondLevelRetries.cs

```diff
@@ -29,9 +40,10 @@ protected internal override void Setup(FeatureConfigurationContext context)
             var retryPolicy = GetRetryPolicy(context.Settings);
 
             context.Container.RegisterSingleton(typeof(SecondLevelRetryPolicy), retryPolicy);
+
             context.Pipeline.Register<SecondLevelRetriesBehavior.Registration>();
 
-            context.Container.ConfigureComponent(b => new SecondLevelRetriesBehavior(retryPolicy, context.Settings.LocalAddress()), DependencyLifecycle.InstancePerCall);
+            context.Container.ConfigureComponent(b => new SecondLevelRetriesBehavior(retryPolicy, context.Settings.LocalAddress(), b.Build<FailureInfoStorage>()), DependencyLifecycle.InstancePerCall);
```

Comment:
This shows that Recoverability should be one feature and not three. But let's address this another day